### PR TITLE
Add the background color setting

### PR DESCRIPTION
**Pull request by umireon**
If we use background removal with this plugin, the background will be black and it is not useful when we want to make the background of the separately recorded video transparent.

スクリーンショット 2022-11-09 23 43 29

With this change, users can specify the background color, and then they can easily apply the Chroma key in post-production.

スクリーンショット 2022-11-09 23 43 48

スクリーンショット 2022-11-09 23 50 18

Technically, this change gives user-specified color to the transparent part of the video.

### CODE REVIEW

File: source-record.c [hunk 1]
```diff
@@ -264,7 +265,10 @@ static void source_record_filter_offscreen_render(void *data, uint32_t cx,
 		return;
 
 	struct vec4 background;
-	vec4_zero(&background);
+	background.x = (double)(filter->backgroundColor & 0xff) / 255.0;
```
Comment (exeldro) [inline source-record.c:268]: why not use the vec4_from_rgba function?

File: source-record.c [hunk 2]
```diff
@@ -842,6 +846,8 @@ static void source_record_filter_update(void *data, obs_data_t *settings)
 		}
 	}
 
+	filter->backgroundColor = obs_data_get_int(settings, "backgroundColor");
```
Comment (exeldro) [inline source-record.c:849]: set the default color in source_record_filter_defaults

**Review by exeldro — CHANGES_REQUESTED** [review]

File: source-record.c (hunk 1 above)
Comment (umireon) [reply source-record.c]: Fixed!

File: source-record.c [hunk 3]
```diff
@@ -1302,6 +1307,14 @@ static obs_properties_t *source_record_filter_properties(void *data)
 	obs_properties_add_group(props, "stream", obs_module_text("Stream"),
 				 OBS_GROUP_NORMAL, stream);
 
+	obs_properties_t *background = obs_properties_create();
+
+	obs_properties_add_color(background, "backgroundColor", obs_module_text("Background Color"));
+
+	obs_properties_add_group(props, "background",
+				 obs_module_text("Background"),
```
[the remark this answers is not in the document]
Comment (umireon) [reply source-record.c:1315]: Added!

**Review by exeldro — APPROVED** [review]

**Pull request merged**